packages/lit-auth-client/src/lib/providers/BaseProvider.ts

```diff
@@ -161,40 +165,43 @@ export abstract class BaseProvider {
 
         let response: SignSessionKeyResponse;
 
+        // common data for the signSessionKey function call
+        const commonData = {
+          sessionKey: sessionKey,
+          statement: authCallbackParams.statement,
+          pkpPublicKey: params.pkpPublicKey,
+          expiration: authCallbackParams.expiration,
+          resources: authCallbackParams.resources,
+          chainId: chainId,
+          ...(params.resourceAbilityRequests && {
+            resourceAbilityRequests: params.resourceAbilityRequests,
+          }),
+        };
+
+        // prepare auth-specific data based on the authentication method
+        let authSpecificData = {};
+
         if (params.authMethod.authMethodType === AuthMethodType.EthWallet) {
-          const authSig = JSON.parse(params.authMethod.accessToken);
-          response = await nodeClient.signSessionKey({
-            statement: authCallbackParams.statement,
-            sessionKey: params.sessionSigsParams.sessionKey,
+          authSpecificData = {
+            authSig: JSON.parse(params.authMethod.accessToken),
             authMethods: [],
-            authSig: authSig,
-            pkpPublicKey: params.pkpPublicKey,
-            expiration: authCallbackParams.expiration,
-            resources: authCallbackParams.resources,
-            chainId,
-
-            // optional
-            ...(params.resourceAbilityRequests && {
-              resourceAbilityRequests: params.resourceAbilityRequests,
-            }),
-          });
+          };
         } else {
-          response = await nodeClient.signSessionKey({
-            sessionKey: params.sessionSigsParams.sessionKey,
-            statement: authCallbackParams.statement,
+          authSpecificData = {
             authMethods: [params.authMethod],
-            pkpPublicKey: params.pkpPublicKey,
-            expiration: authCallbackParams.expiration,
-            resources: authCallbackParams.resources,
-            chainId,
-
-            // optional
-            ...(params.resourceAbilityRequests && {
-              resourceAbilityRequests: params.resourceAbilityRequests,
-            }),
-          });
+          };
         }
 
+        // Merge the common and auth-specific data
+        response = await nodeClient.signSessionKey({
+          // default
+          authMethods: [],
+
+          // override
+          ...commonData,
+          ...authSpecificData,
```

Comment:
While combining the common properties into a single `commonData` is awesome and reduces cognitive load here, I think `authSpecificData` as a dedicated variable does the opposite because it is entirely different depending on the code that has run before this line.  

It would be easier to follow if we kept the existing pattern of composing the arguments to send to `signSessionKey()` as separate blocks and didn't define an intermediary `authSpecificData` that is a different shape of object depending on the code above it... this way there is an extra layer of indirection that we need to follow to figure out what `authSpecificData` will be and what we're passing into this method.

After in-lining this, we can also get rid of the `// default` entry with an empty `authMethods` array; it's actually duplicative given that we define it as an empty array on L187 anyway :)